runtime/mini-mock/src/hooks.rs

```diff
@@ -1,30 +1,59 @@
 use frame_support::weights::Weight;
 
 use crate::{MiniRuntime as Runtime, *};
-use sp_runtime::traits::One;
+use sp_runtime::traits::{One, Zero};
 
 pub struct GlobalOnInitQueues;
 
-impl pallet_clock::traits::OnHookQueues<Runtime> for GlobalOnInitQueues {
+impl t3rn_primitives::clock::OnHookQueues<Runtime> for GlobalOnInitQueues {
     fn process(n: BlockNumber, on_init_weight_limit: Weight) -> Weight {
         let mut weights_consumed = vec![];
-        const PROCESS_SIGNAL_SHARE: u32 = 15;
-        const XTX_TICK_SHARE: u32 = 35;
-        const REVERT_XTX_SHARE: u32 = 35;
+        const PROCESS_SIGNAL_SHARE: u32 = 5;
+        const XTX_TICK_SHARE: u32 = 30;
+        const REVERT_XTX_SHARE: u32 = 5;
         const BUMP_ROUND_SHARE: u32 = 5;
-        const CALC_CLAIMABLE_SHARE: u32 = 10;
+        const CALC_CLAIMABLE_SHARE: u32 = 5;
+        const WEEKLY_SHARE: u32 = 20;
+        const DAILY_SHARE: u32 = 10;
+        const HOURLY_SHARE: u32 = 20;
+
         if PROCESS_SIGNAL_SHARE
             + XTX_TICK_SHARE
             + REVERT_XTX_SHARE
             + BUMP_ROUND_SHARE
             + CALC_CLAIMABLE_SHARE
+            + WEEKLY_SHARE
+            + DAILY_SHARE
+            + HOURLY_SHARE
             > 100
         {
             log::error!(
                 "GlobalOnInitQueues::Invalid shares exceed 100%, returning 0 - re-check the shares"
             );
             return 0
         }
+        const BLOCKS_PER_WEEK: BlockNumber = 7 * 24 * 60 * 60;
+        const BLOCKS_PER_DAY: BlockNumber = 24 * 60 * 60;
+        const BLOCKS_PER_HOUR: BlockNumber = 60 * 60;
+
+        if (n % BLOCKS_PER_HOUR).is_zero() {
+            let hourly_weight_limit: Weight =
+                Perbill::from_percent(HOURLY_SHARE).mul_ceil(on_init_weight_limit);
```

Comment:
Do we care to know if it saturates when multiplying?

Reply:
na it taking percentage should be less than weight limit, saturating is complex here due to types: saturating_mul function of the Perbill type expects another Perbill as an argument, but on_init_weight_limit is a u64 in your code.